BranchSDK-Samples/Windows/TestBed/TextField.cpp

```diff
@@ -72,10 +72,10 @@ TextField::appendText(const std::wstring& text, size_t maxLength)
 	ScopeLock l(m_lock);
 	wstring newText = getText() + L"\r\n" + text;
 	// Limit text length
-	const wstring::size_type total = newText.length();
-	const __int64 offset = total - maxLength;
```

Comment:
The reason this was failing is it was taking the difference of two unsigned integers. The return type of that operation is the same as the operands, so unsigned. Only after computing this large, positive value is it converted to a signed integer, but too late. Rather than casting both args here, a direct comparison (total > maxLength) is safer.